### PR TITLE
Make the data types of output and input consistenst for batchnorm

### LINKED ISSUE #86790: RuntimeError caused by batchnorm
### 🐛 Describe the bug

The model TTS will crash due to the issue: when input of BN is not contiguous and the data type of input is different with that of parameters, BN will raise error RuntimeError: !needs_dynamic_casting<func_t>::check(iter) INTERNAL ASSERT FAILED at "xxx/pytorch/aten/src/ATen/native/cpu/Loops.h":311, please report a bug to PyTorch.

### Versions

PyTorch master and PyTorch 1.13
python 3.8.5

### PR DESCRIPTION
**Pull request by CaoE**
The model TTS will crash due to the issue:: when input of BN is not contiguous and the data type of input is different with that of parameters, BN will raise error `RuntimeError: !needs_dynamic_casting<func_t>::check(iter) INTERNAL ASSERT FAILED at "xxx/pytorch/aten/src/ATen/native/cpu/Loops.h":311, please report a bug to PyTorch`.

Make the data types of output and input consistenst for batchnorm to fix the issue.

### CODE REVIEW

**Review by mingfeima — APPROVED** [review]

**Comment by CaoE** [comment]
@frank-wei  Could you please view this PR ? Thank you.

**Review by jgong5 — APPROVED** [review]

**Comment by CaoE** [comment]
@malfet  Could you please view this PR ?  It will fix a crash issue. Thank you.

File: test/test_nn.py [hunk 1]
```diff
@@ -8843,6 +8843,15 @@ def test_batchnorm_non_contig_cpu(self):
         self.assertTrue(ref_out.is_contiguous())
         self.assertEqual(out, ref_out)
 
+        input_bf = torch.arange(24, dtype=torch.bfloat16).reshape(1, 3, 2, 4).cpu()
```
Comment (malfet) [inline test/test_nn.py:8846]: Hmm, why CPU is needed here?
```suggestion
        input_bf = torch.arange(24, dtype=torch.bfloat16).reshape(1, 3, 2, 4)
```

File: test/test_nn.py [hunk 2]
```diff
@@ -8843,6 +8843,15 @@ def test_batchnorm_non_contig_cpu(self):
...
+        input_bf = input_bf.permute(0, 2, 1, 3)
+        input_f = input_bf.float()
+        bn_mix = torch.nn.BatchNorm2d(2).cpu().float().eval()
```
Comment (malfet) [inline test/test_nn.py:8849]: Same as above, why cast to cpu here?
```suggestion
        bn_mix = torch.nn.BatchNorm2d(2).float().eval()
```

**Review by malfet — APPROVED** [review]
LGTM, would be nice to answer the question about explicit `.cpu()` calls

**Comment by malfet** [comment]
Also, please note that `RuntimeError:` is not a crash, but a structured exception, that could be safely handled by callee.

**Comment by CaoE** [comment]
@malfet  Thank you for your correction and kind help. May I know if this PR will be landed in pytorch 1.13 ?

**Pull request closed without merging**